pc_software/lib/jd_beacon/enumerators/udev_enumerator.rb

```diff
@@ -100,11 +100,13 @@ def find_matching_udev_devices(&block)
       # attached beacon board Abstract Control Model USB-to-serial devices.
       #
       def connected_beacon_board_udev_devices
-        find_matching_udev_devices do |enumerator|
+        devices = find_matching_udev_devices do |enumerator|
           enumerator.match_subsystem("tty")
           enumerator.match_property("ID_VENDOR_ID", VENDOR_ID)
           enumerator.match_property("ID_MODEL_ID", PRODUCT_ID)
         end
+
+        devices.sort { |a,b| a.property("ID_SERIAL") <=> b.property("ID_SERIAL") }
```

Comment:
I can see why you're doing this here-- but I'm not convinced this is the best place for it. If you do want to leave it here, I'd suggest at least commenting on _why_ that's desirable.  (There's nothing worse than going into a function like this and having _no idea_ why someone would want to sort the devices.)

Either way, I'd prefer using `sort_by` to the block you've put in:

```ruby
devices.sort_by { |d| d.property("ID_SERIAL") }
```

This:
1. Look a lot nicer; and
2. Automatically memoizes calls to property, avoiding the foreign function interface calls.